config/backend-config/backend-config.go

```diff
@@ -193,37 +189,38 @@ Available topics are:
 - TopicProcessConfig: Will receive only backend configuration of processor enabled destinations
 - TopicRegulations: Will receive all regulations
 */
-func (bc *CommonBackendConfig) Subscribe(ctx context.Context, topic Topic) pubsub.DataChannel {
+func (bc *commonBackendConfig) Subscribe(ctx context.Context, topic Topic) pubsub.DataChannel {
 	return bc.eb.Subscribe(ctx, string(topic))
 }
 
 func newForDeployment(deploymentType deployment.Type, configEnvHandler types.ConfigEnvI) (BackendConfig, error) {
-	var backendConfig BackendConfig
+	backendConfig := &commonBackendConfig{
+		eb: pubsub.New(),
+	}
+	parsedConfigBackendURL, err := url.Parse(configBackendURL)
+	if err != nil {
+		return nil, fmt.Errorf("invalid config backend URL: %v", err)
+	}
 
 	switch deploymentType {
 	case deployment.DedicatedType:
-		backendConfig = &SingleWorkspaceConfig{
-			CommonBackendConfig: CommonBackendConfig{
-				configEnvHandler: configEnvHandler,
-				eb:               pubsub.New(),
-			},
+		backendConfig.workspaceConfig = &singleWorkspaceConfig{
+			configJSONPath:   configJSONPath,
+			configBackendURL: parsedConfigBackendURL.String(),
+			configEnvHandler: configEnvHandler,
 		}
 	case deployment.MultiTenantType:
 		isNamespaced := config.IsEnvSet("WORKSPACE_NAMESPACE")
 		if isNamespaced {
-			backendConfig = &NamespaceConfig{
-				CommonBackendConfig: CommonBackendConfig{
-					configEnvHandler: configEnvHandler,
-					eb:               pubsub.New(),
-				},
+			backendConfig.workspaceConfig = &namespaceConfig{
+				ConfigBackendURL: parsedConfigBackendURL,
```

Comment:
Any reason why this struct has ConfigBackendURL of type *url.URL and other two structs (singleWorkspaceConfig & multiTenantWorkspacesConfig) have configBackendURL of type string?

Reply:
It was born that way (the namespace config) and I didn't refactor the other two to be consistent. I'll have a look and see if I can refactor them easily once we merge @lvrach [PR](https://github.com/rudderlabs/rudder-server/pull/2269). I expect some conflicts upon merging, so I'll have to do some more work here anyway.

Reply:
@chandumlg it was super easy, it's done now :+1: